spec/03-plugins/statsd/log_spec.lua

```diff
@@ -0,0 +1,246 @@
+        port = UDP_PORT, 
+      }
+    })
+
+    -- plugin 6
+    assert(helpers.dao.plugins:insert {
+      api_id = api6.id,
+      name = "statsd", 
+      config = { 
+        host = "127.0.0.1", 
+        port = UDP_PORT, 
+        metrics = {"response_size"}
+      }
+    })
+
+    assert(helpers.start_kong())
+  end)
+
+  teardown(function()
+    if client then client:close() end
+    helpers.stop_kong()
+  end)
+
+  before_each(function()
+    client = assert(helpers.http_client("127.0.0.1", helpers.test_conf.proxy_port))
+  end)
+  
+  after_each(function()
+    if client then client:close() end
+  end)
+
+
+  it("should log to UDP when metrics is request_count", function()
+    local thread = helpers.udp_server(UDP_PORT) -- Starting the mock UDP server
+
+    local response = assert( client:send {
+        method = "GET",
+        path = "/request",
+        headers = {
+          host = "logging1.com"
+        }
+      })
+
+    assert.has.res_status(200, response)
+
+    local ok, res = thread:join()
+    assert.True(ok)
+    assert.truthy(res)
+    assert.equal("kong.logging1_com.request.count:1|c", res)
+  end)
+
+  it("should log to UDP when metrics is status_count", function()
+    local thread = helpers.udp_server(UDP_PORT) -- Starting the mock UDP server
+
+    local response = assert( client:send {
+        method = "GET",
+        path = "/request",
+        headers = {
+          host = "logging3.com"
+        }
+      })
+
+    assert.has.res_status(200, response)
+
+    local ok, res = thread:join()
+    assert.True(ok)
+    assert.truthy(res)
+    assert.equal("kong.logging3_com.request.status.200:1|c", res)
+  end)
+
+  it("should log to UDP when metrics is request_size", function()
+    local thread = helpers.udp_server(UDP_PORT) -- Starting the mock UDP server
+
+    local response = assert( client:send {
+        method = "GET",
+        path = "/request",
+        headers = {
+          host = "logging4.com"
+        }
+      })
+
+    assert.has.res_status(200, response)
+
+    local ok, res = thread:join()
+    assert.True(ok)
+    assert.truthy(res)
+    local message = {}
+    for w in string.gmatch(res,"kong.logging4_com.request.size:%d*|g") do
+      table.insert(message, w)
+    end
+    assert.equal(1, #message)
+  end)
+
+  it("should log to UDP when metrics is latency", function()
+    local thread = helpers.udp_server(UDP_PORT) -- Starting the mock UDP server
+
+    local response = assert( client:send {
+        method = "GET",
+        path = "/request",
+        headers = {
+          host = "logging2.com"
+        }
+      })
+
+    assert.has.res_status(200, response)
+
+    local ok, res = thread:join()
+    assert.True(ok)
+    assert.truthy(res)
+    local message = {}
+    for w in string.gmatch(res,"kong.logging2_com.latency:.*|g") do
+      table.insert(message, w)
+    end
+
+    assert.equal(1, #message)
+  end)
+
+  it("should log to UDP when metrics is request_count", function()
```

Comment:
The point of those tests refactoring is to also improve the quality of our tests (not just convert to a new tooling). Better messages, titles, assertions, leaner tests etc...

Here, we already have the exact same test name (the first one). I believe the correct name for this test is "defaults to `request_count` metric" according to the plugin configuration fixture in `setup()`.

Secondly, it is my feeling this entire set of instructions:
```lua
assert.truthy(res)
local message = {}
for w in string.gmatch(res,"kong.logging2_com.latency:.*|g") do
table.insert(message, w)
end

assert.equal(1, #message)
```

Can be replaced with:
```lua
assert.matches("kong.logging2_com.latency:.*|g", res)
```